ot/da.py

```diff
@@ -843,10 +927,896 @@ def predict(self, x):
+            The class labels
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+
+        # pairwise distance
+        self.cost_ = dist(Xs, Xt, metric=self.metric)
+
+        if (ys is not None) and (yt is not None):
+
+            if self.limit_max != np.infty:
+                self.limit_max = self.limit_max * np.max(self.cost_)
+
+            # assumes labeled source samples occupy the first rows
+            # and labeled target samples occupy the first columns
+            classes = np.unique(ys)
+            for c in classes:
+                idx_s = np.where((ys != c) & (ys != -1))
+                idx_t = np.where(yt == c)
+
+                # all the coefficients corresponding to a source sample
+                # and a target sample :
+                # with different labels get a infinite
+                for j in idx_t[0]:
+                    self.cost_[idx_s[0], j] = self.limit_max
+
+        # distribution estimation
+        self.mu_s = self.distribution_estimation(Xs)
+        self.mu_t = self.distribution_estimation(Xt)
+
+        # store arrays of samples
+        self.Xs = Xs
+        self.Xt = Xt
+
+        return self
+
+    def fit_transform(self, Xs=None, ys=None, Xt=None, yt=None):
+        """Build a coupling matrix from source and target sets of samples
+        (Xs, ys) and (Xt, yt) and transports source samples Xs onto target
+        ones Xt
+
+        Parameters
+        ----------
+        Xs : array-like, shape (n_source_samples, n_features)
+            The training input samples.
+        ys : array-like, shape (n_source_samples,)
+            The class labels
+        Xt : array-like, shape (n_target_samples, n_features)
+            The training input samples.
+        yt : array-like, shape (n_labeled_target_samples,)
+            The class labels
+
+        Returns
+        -------
+        transp_Xs : array-like, shape (n_source_samples, n_features)
+            The source samples samples.
+        """
+
+        return self.fit(Xs, ys, Xt, yt).transform(Xs, ys, Xt, yt)
+
+    def transform(self, Xs=None, ys=None, Xt=None, yt=None, batch_size=128):
+        """Transports source samples Xs onto target ones Xt
+
+        Parameters
+        ----------
+        Xs : array-like, shape (n_source_samples, n_features)
+            The training input samples.
+        ys : array-like, shape (n_source_samples,)
+            The class labels
+        Xt : array-like, shape (n_target_samples, n_features)
+            The training input samples.
+        yt : array-like, shape (n_labeled_target_samples,)
+            The class labels
+        batch_size : int, optional (default=128)
+            The batch size for out of sample inverse transform
+
+        Returns
+        -------
+        transp_Xs : array-like, shape (n_source_samples, n_features)
+            The transport source samples.
+        """
+
+        if np.array_equal(self.Xs, Xs):
```

Comment:
```self.Xs is Xs ``` will test if the array is the same object and avoid the multiple tests and computation in np_array_equal.